### ISSUE TITLE
Rubocop Autochange on `if !!something`

### ISSUE THREAD

**Issue by deniciocode** [rubocop/rubocop#10583]
## Is your feature request related to a problem? Please describe.

After running `rubocop -A` on my controller the following change was made an resulted in failing specs.
```ruby
# original code
if !!raise_error

# changed by rubocop
unless raise_error.nil?
```
This is dangerous if we have `booleans`. 

## Describe the solution you'd like
To avoid the error above I do not have a suggestion in pure ruby. 
In rails I go with:
```ruby
if rais_error.present?
```
This solved the issue for me.

**Comment by bbatsov** [comment]
If it's a boolean the double-negation doesn't make a lot of sense in the first place. 😉 I think the auto-correct is already marked as unsafe. There's little else we can do on this topic.

**Comment by jonas054** [comment]
Agreed. Note that `rubocop -A` performs all autocorrections, safe and unsafe. Closing.
